ocaml/xapi-idl/lib/posix_channel.ml

```diff
@@ -80,12 +80,17 @@ let proxy (a : Unix.file_descr) (b : Unix.file_descr) =
       in
       (* If we can't make any progress (because fds have been closed), then stop *)
       if r = [] && w = [] then raise End_of_file ;
-      let r, w, _ = Unix.select r w [] (-1.0) in
-      (* Do the writing before the reading *)
-      List.iter
-        (fun fd -> if a = fd then CBuf.write b' a else CBuf.write a' b)
-        w ;
-      List.iter (fun fd -> if a = fd then CBuf.read a' a else CBuf.read b' b) r ;
+      let epoll = Polly.create () in
+      List.iter (fun fd -> Polly.add epoll fd Polly.Events.inp) (r @ w) ;
```

Comment:
I think you need `Polly.Events.out` for `w` to wait for it to be ready for writing.

Reply:
Good point, although we need to read and write: I guess there needs to be two Polly.adds?

Reply:
I believe you need to create a set of all events you care about; using `++` in `Polly.Events`.